library/alloc/src/vec.rs

```diff
@@ -2199,19 +2199,12 @@ impl<T> SpecFromIter<T, IntoIter<T>> for Vec<T> {
         // But it is a conservative choice.
         let has_advanced = iterator.buf.as_ptr() as *const _ != iterator.ptr;
         if !has_advanced || iterator.len() >= iterator.cap / 2 {
-            unsafe {
-                let it = ManuallyDrop::new(iterator);
-                if has_advanced {
-                    ptr::copy(it.ptr, it.buf.as_ptr(), it.len());
-                }
-                return Vec::from_raw_parts(it.buf.as_ptr(), it.len(), it.cap);
-            }
+            // Safety: passing 0 is always valid
+            return unsafe { iterator.into_vec_with_uninit_prefix(0) };
```

Comment:
I think it might be clearer to add another helper method that just converts the iterator into a `Vec` without talking about an uninitialized prefix here.

Reply:
Looking at the implementation of `into_vec_with_uninit_prefix` I think it would be better if we disallowed a prefix of `0` and added a separate `into_vec` method.

Reply:
As commented below, shifting is still needed even if the prefix is 0 so the methods would be identical except for their safety and one less addition.

Reply:
I personally think that's enough reason to make the change.